src/test/java/org/humancellatlas/ingest/file/FileTest.java

```diff
@@ -69,4 +73,19 @@ public void testAddToAnalysisWhenFileAlreadyLinkedToSubmissionEnvelope() {
         assertThat(file.getDerivedByProcesses()).contains(process);
         assertThat(file.getSubmissionEnvelope()).isEqualTo(submissionEnvelope);
     }
+
+    @ParameterizedTest
+    @MethodSource("testFiles")
+    public void newFileHasDataFileUuidNotNull(File file) {
+        assertThat(file)
+                .extracting("dataFileUuid")
+                .doesNotContainNull();
+    }
+
+    private static Stream<File> testFiles() {
+        return Stream.of(
+                new File(),
+                new File(null, "test-File")
+        );
+    }
```

Comment:
This is brilliant, it's been bugging me for ages that JUnit doesn't allow complex parameterised tests and this is a great solution.

Reply:
Junit has allowed parameterised tests since the early versions, but really improved the API in version 5.